conanfile.py

```diff
@@ -8,53 +14,86 @@ class CatchConan(ConanFile):
+        shutil.copy2(f'{self.recipe_folder}/LICENSE.txt', f'{self.package_folder}/licenses/')
+
     def package_info(self):
         lib_suffix = "d" if self.settings.build_type == "Debug" else ""
 
-        self.cpp_info.names["cmake_find_package"] = "Catch2"
-        self.cpp_info.names["cmake_find_package_multi"] = "Catch2"
-        # Catch2
-        self.cpp_info.components["catch2base"].names["cmake_find_package"] = "Catch2"
-        self.cpp_info.components["catch2base"].names["cmake_find_package_multi"] = "Catch2"
-        self.cpp_info.components["catch2base"].names["pkg_config"] = "Catch2"
-        self.cpp_info.components["catch2base"].libs = ["Catch2" + lib_suffix]
-        self.cpp_info.components["catch2base"].builddirs.append("lib/cmake/Catch2")
-        # Catch2WithMain
-        self.cpp_info.components["catch2main"].names["cmake_find_package"] = "Catch2WithMain"
-        self.cpp_info.components["catch2main"].names["cmake_find_package_multi"] = "Catch2WithMain"
-        self.cpp_info.components["catch2main"].names["pkg_config"] = "Catch2WithMain"
-        self.cpp_info.components["catch2main"].libs = ["Catch2Main" + lib_suffix]
-        self.cpp_info.components["catch2main"].requires = ["catch2base"]
+        if conan_version < scm.Version("2.0.0"):
+            self.cpp_info.names["cmake_find_package"] = "Catch2"
+            self.cpp_info.names["cmake_find_package_multi"] = "Catch2"
+            # Catch2
+            self.cpp_info.components["catch2base"].names["cmake_find_package"] = "Catch2"
+            self.cpp_info.components["catch2base"].names["cmake_find_package_multi"] = "Catch2"
+            self.cpp_info.components["catch2base"].names["pkg_config"] = "Catch2"
+            self.cpp_info.components["catch2base"].libs = ["Catch2" + lib_suffix]
+            self.cpp_info.components["catch2base"].builddirs.append("lib/cmake/Catch2")
+            # Catch2WithMain
+            self.cpp_info.components["catch2main"].names["cmake_find_package"] = "Catch2WithMain"
+            self.cpp_info.components["catch2main"].names["cmake_find_package_multi"] = "Catch2WithMain"
+            self.cpp_info.components["catch2main"].names["pkg_config"] = "Catch2WithMain"
+            self.cpp_info.components["catch2main"].libs = ["Catch2Main" + lib_suffix]
+            self.cpp_info.components["catch2main"].requires = ["catch2base"]
+        else:
+            self.cpp_info.set_property("cmake_file_name", "Catch2")
+
+            # Catch2
+            self.cpp_info.components["catch2base"].set_property("cmake_file_name", "Catch2")
+            self.cpp_info.components["catch2base"].set_property("pkg_config_name", "Catch2")
+            self.cpp_info.components["catch2base"].libs = ["Catch2" + lib_suffix]
+            self.cpp_info.components["catch2base"].builddirs.append("lib/cmake/Catch2")
+
+            # Catch2WithMain
+            self.cpp_info.components["catch2main"].set_property("cmake_file_name", "Catch2WithMain")
```

Comment:
Please do specify `cmake_target_name` similar to
https://github.com/conan-io/conan-center-index/blob/master/recipes/catch2/3.x.x/conanfile.py#L168

That way you would be able to correctly use `Catch2::Catch2WithMain` target instead of the lowercase.

Reply:
Should be updated. Does this not fix the issue then with the linked issue? I didn't realize that the conan file was seperate in the conan.io repo.

Reply:
I think i did this. I kept the `cmake_file_name` but not sure if that matters. 

I also didn't realize that the conan file for conan.io is separate from this repo. So will this close the issue in reference? Maybe if it is maintained over there we just drop the conan files here

Reply:
The Conan Center policy is to maintain recipes for the default repo so in-source conanfiles are only needed for project-specific development. I'm not sure how many people require it so it's up to the maintainer in here to decide if it's still needed. 

Regarding this specific PR - `cmake_target_name` is used for the target name. It needs to be specified with namespace because of the different filename to namespace to target to component conventions different projects have. `cmake_file_name` is needed only for the `find_package` syntax, it's not useful for components and it doesn't make sense to specify the target name in there. The global level `cmake_target_name` I think refers to the target which is wired on Conan-level global component dependencies, it's not actually used in CMake. 

Additionally, prefer `os.path.join` over the `/` directory separator for cross-environment compatibility.